packages/app-page-layer/src/PageTree.tsx

```diff
@@ -1,36 +1,101 @@
-import { useCallback } from "react";
+import { useCallback, useMemo } from "react";
 import { PageInfo } from "./types";
 import { canvasApi } from "@atrilabs/pwa-builder-manager";
+import { Tree } from "antd";
+import type { DataNode, DirectoryTreeProps } from "antd/es/tree";
+import { gray400, gray700 } from "@atrilabs/design-system";
+
+const { DirectoryTree } = Tree;
 
 export const PageTree: React.FC<{
   pagesInfo: PageInfo;
   selectedPageRouteObjectPath: string;
-  onCloseClicked: () => void;
 }> = (props) => {
-  const onPageClicked = useCallback((urlPath: string) => {
-    canvasApi.navigatePage(urlPath);
-  }, []);
+  const createTreeNodes = useCallback(
```

Comment:
path - "/countries/usa.tsx"
pathArr - ["countries", "usa.tsx"]
treeNodes - for reference 

1. Create a leaf node with the last element
2. Traverse and check if path is present, if not create it else traverse it further
3. Append the node to the desired location